addons/web_editor/models/ir_ui_view.py

```diff
@@ -80,10 +85,66 @@ def save_oe_structure(self, el):
             'mode': 'extension',
         }
         vals.update(self._save_oe_structure_hook())
-        self.env['ir.ui.view'].create(vals)
+        oe_structure_view = self.env['ir.ui.view'].create(vals)
+        self._copy_custom_snippet_translations(oe_structure_view, 'arch_db')
 
         return True
 
+    @api.model
+    def _copy_custom_snippet_translations(self, record, html_field):
+        """ Given a ``record`` and its HTML ``field``, detect any
+        usage of a custom snippet and copy its translations.
+        """
+        lang_value = record[html_field]
+        if not lang_value:
+            return
+
+        tree = html.fromstring(lang_value)
+        for custom_snippet_el in tree.xpath('//*[hasclass("s_custom_snippet")]'):
+            custom_snippet_name = custom_snippet_el.get('data-name')
+            custom_snippet_view = self.search([('name', '=', custom_snippet_name)], limit=1)
+            if custom_snippet_view:
+                self._copy_html_field_translations(custom_snippet_view, 'arch_db', record, html_field)
+
+    @api.model
+    def _copy_html_field_translations(self, record_from, field_from, record_to, field_to):
+        record_from.check_access_rights('read')
+        record_from.check_access_rule('read')
+        record_to.check_access_rights('write')
+        record_to.check_access_rule('write')
+
+        lang_value = record_from[field_from]
+        if not lang_value:
+            return
+
+        lang_env = self.env.lang or 'en_US'
+        langs = set(lang for lang, _ in self.env['res.lang'].get_installed())
+
+        # 1. Get translations
+        record_from.flush_model([field_from])
+        query = sql.SQL(
+            """SELECT {column} FROM {table} WHERE id = %(record_id)s"""
+        ).format(
+            table=sql.Identifier(record_from._table),
+            column=sql.Identifier(field_from),
+        )
+        self.env.cr.execute(query, {
+            'record_id': record_from.id,
+        })
+        field_db = self.env.cr.fetchone()[0]
+        translation_dictionary = record_from._fields[field_from].get_translation_dictionary(
+            field_db.get(lang_env) or field_db.get('en_US'),
+            {lang: value for lang, value in field_db.items() if lang in langs and lang != lang_env}
+        )
+
+        # 2. Set translations
+        field = record_to._fields[field_to]
+        new_value = {
+            lang: field.translate(lambda term: translation_dictionary.get(term, {lang: None}).get(lang), record_to[field_to])
+            for lang in field_db.keys()
+        }
+        record_to.env.cache.update_raw(record_to, field, [new_value], dirty=True)
```

Comment:
`update_raw` is not enough.
1. we should at least trigger the `record_to.modified([field_to])` to recompute everything depends on the `field_to`
2. triggering the override of `write` may be necessary